### ISSUE TITLE
Softmax layer not correct

### ISSUE THREAD

**Issue by galla15** [fastmachinelearning/hls4ml#1431]
Hi, I’m trying to compile a pretrained model from
https://huggingface.co/aman-27/cnn-digit-recognition-tf-v1/tree/main

but I’m encountering an error during the compilation stage.

The last layer, which should be a Softmax layer, is generated as an Activation layer. When compilation starts, it complains about missing attributes in the config struct generated by hls4ml.

I tried manually adding the missing components to the C struct (parameters.h), but I quickly end up in a rabbit hole of manual modifications just to get it to compile, which defeats the purpose.

Has anyone else experienced this behavior with Softmax?

To reproduce the issue, simply download the *.keras file from Hugging Face and try to build it.

My configuration is:

Backend: Vitis 2024.1

Precision: fixed<16,6>

Reuse factor: 1

Strategy: Latency

**Comment by bo3z** [comment]
Thank you for the code. Can you also post the exact compilation error you're getting? Looking at the above, the generated struct looks correct.

Also, note, hls4ml doesn't add a softmax layer - the softmax layer must already be present in the input model (it's probably part of the output dense layer, with activation = 'softmax')

**Comment by galla15** [comment]
Hi @bo3z , it seems to solve the issue, at least with the 2 models that I was trying to compile. Thanks !
